### PR TITLE
Create experimental test optics_thresholds_test.go

### PR DESCRIPTION
**Pull request by sezhang2**
Validate optics high and low thresholds for input power, output power, temperature and bias-current.

### CODE REVIEW

**Review by bstoll — APPROVED** [review]

File: feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+					t.Errorf("Get threshold for %q): got %.2f, want within [%f, %f] ", transceiver, threshold, tc.minThreshold, tc.maxThreshold)
+				}
+			}
+		})
+	}
+}
+
+func fetchOpticsThreshold(t *testing.T, dut *ondatra.DUTDevice, opticsName string, property string) float64 {
+	t.Helper()
+	// TODO: Need to update the lookup code after optics threshold model is defined.
```
Comment (wenovus) [inline feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go:200]: TODOs are better with a username or an issue number.

File: feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+		return float64(v)
+	case telemetry.UnionFloat64:
+		return float64(v)
+	default:
+		t.Fatalf("Error extracting optics threshold, could not type assert union. union: %v", val)
+		return 0
+	}
+}
+
+func findComponentsByType(t *testing.T, dut *ondatra.DUTDevice, cType telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT) []string {
```
Comment (wenovus) [inline feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go:217]: (non-blocking) Is this a duplicate from somewhere else? If so, should there be a common internal library for doing this?

File: feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+	t.Logf("Found transceiver list: %v", transceivers)
+	if len(transceivers) == 0 {
+		t.Fatalf("Get transceiver list for %q: got 0, want > 0", dut.Model())
+	}
+
+	cases := []struct {
+		desc         string
+		property     string
+		minThreshold float64
+		maxThreshold float64
```
Comment (wenovus) [inline feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go:93]: wantMinThreshold, wantMaxThreshold

**Review by wenovus — APPROVED** [review]

File: feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go (hunk 1 above)
Comment (sezhang2) [reply feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go:200]: I removed it per our internal requirement.

File: feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go (hunk 2 above)
Comment (sezhang2) [reply feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go:217]: Yes, we are working on helper functions.

**Pull request merged**